In [1]:
import numpy as np
import pandas as pd
import os

# NULL HYPOTHESIS:
## The average running time of the new bus route is the same or higher than the average running time of the original bus route.
# _$H_0$_ : $T_{\mathrm{original}} <= {T_{\mathrm{new}}}$
# _$H_1$_ : $T_{\mathrm{original}} > {T_{\mathrm{new}}}$

## I will use a significance level  $\alpha=0.05$

#### which means i want the probability of getting a result at least as significant as mine to be less then 5%

In [21]:
data = pd.read_csv('times.txt', sep = " ", header = None)

In [22]:
data.head()

0
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

In [23]:
mymean = float(data.mean())
mystd = float(data.std())
N = len(data)

In [24]:
miu_pop = 36.0
std_pop = 6.0

In [25]:
Z = (miu_pop - mymean)/(std_pop/np.sqrt(N))
print(Z)
print(type(Z))

2.55639718617
<class 'numpy.float64'>


## According to the Z-score table, we can get the corresponsing p-value for Z = 2.556397 is 0.00529, which is smaller than 0.05.
## Thus we can conclude that the null hypothesis is rejected.